modules/core/src/main/java/org/apache/ignite/spi/encryption/keystore/KeystoreEncryptionSpi.java

```diff
@@ -113,24 +113,13 @@
      */
     private int keySize = DEFAULT_KEY_SIZE;
 
-    /**
-     * Master key name.
-     */
-    private String masterKeyName = DEFAULT_MASTER_KEY_NAME;
-
-    /**
-     * Master key.
-     */
-    private KeystoreEncryptionKey masterKey;
+    /** Tuple of the master key id and the master key. */
+    private volatile T2<String, KeystoreEncryptionKey> masterKey = new T2<>(DEFAULT_MASTER_KEY_NAME, null);
```

Comment:
Not clear to me.

Why do we need volatile guarantees here?

Reply:
I have implemented two variables.
But synchronization not needed because `EncryptionManager` guarantee that no one key encryption method will be called during the `setMasterKeyName` executes.